src/controller/LoginController.php

```diff
@@ -2,43 +2,50 @@
 
 class LoginController {
 
+    const COOKIE_EXPIRE_TIME = 3600;
+
     private $connection;
 
     function __construct($connection) {
         $this->connection = $connection;
     }
 
     public function loginAction() {
-        if (isset($_POST['userName']) && isset($_POST['password'])) {
+
+        if (empty($_POST['userName']) && empty($_POST['password'])) {
+            include("views/login.php");
+        } else {
             $userName = $_POST['userName'];
             $password = $_POST['password'];
             $status = $this->userExist($userName, $password);
 
-            if ($status == 1) {
-                setcookie("login", 1, time() + 60, "/");
-                header("Location: index.php?controller=welcome&action=welcome");
-            } elseif ($status == 2) {
-                setcookie("error", "Klaida, prisijungti nepavyko ", time() + 15, "/");
+            if ($status == FALSE) {
+                include("views/error.php");
+            } elseif ($status == TRUE) {
+                setcookie("login", 'login', time() + self::COOKIE_EXPIRE_TIME, "/");
+                $_GET['controller'] = "welcome";
+                $_GET['action'] = "welcome";
+                header("Location: index.php");
             }
-        } else {
-            include("views/login.php");
         }
     }
 
     public function logoutAction() {
         if (isset($_POST['logout'])) {
-            setcookie("login", 1, time() - 60, "/");
-            header("Location: index.php?controller=login&action=login");
+            setcookie("login", 'logout', time() - self::COOKIE_EXPIRE_TIME, "/");
+            $_GET['controller'] = "login";
+            $_GET['action'] = "login";
+            header("Location: index.php");
         }
     }
 
     private function userExist($userName, $password) {
         $sqlQuery = "SELECT * FROM users WHERE userName='$userName' AND password='$password'";
         $result = $this->connection->query($sqlQuery);
         if ($result->num_rows > 0) {
-            return $status = 1;
+            return $status = TRUE;
```

Comment:
Papasakok kam šioje vietoje reikalingas `$status` kintamasis

Reply:
Nu, kaip aukščiau nurodei, status indentifikavo ar rastas vartotojo prisijungimo vardas ir slaptažodis duombazėje, ir tą rezutatą saugodavau į kintamąjį, dabar visai tai pašalinau, nes tai nebuvo reikalinga (veikia ir be to).

Reply:
`return $status = TRUE;` negrąžins kintamojo $status reikšmės. Visuomet bus grąžinama TRUE. Ar žinai kodėl? `The value of an assignment expression is the value assigned.` (https://www.php.net/manual/en/language.operators.assignment.php). Dėl to galima daryti pvz: `$a = $b = $c = 0`